apps/costcontrol/test/unit/startup_test.js

```diff
@@ -404,50 +404,49 @@ suite('Application Startup Modes Test Suite >', function() {
     CostControlApp.init();
   });
 
-  test(
-    'DSDS Ensure the FTE will be closed when there are a data slot change',
+  test('UpdateUI always execute the callback function with a mode change',
     function(done) {
-      var defaultDataSlotId = 0, newDataSlotId = 1,
-          dataSlot = 'ril.data.defaultServiceId';
-
-      MockNavigatorSettings.mSettings[dataSlot] = defaultDataSlotId;
       var applicationMode = 'DATA_USAGE_ONLY';
       setupCardState({cardState: 'ready'});
-      window.Common = new MockCommon({ activateFTEListener: true });
       window.ConfigManager = new MockConfigManager({
         fakeSettings: { fte: true },
         applicationMode: applicationMode
       });
-
-      MockNavigatorMozMobileConnections.mAddMobileConnection(
-        {  iccId: '0000000' } ,1);
-
-      SimManager.reset();
-
-      window.addEventListener('ftestarted', function _onftestarted(evt) {
-        window.removeEventListener('ftestarted', _onftestarted);
-        var iframe = document.getElementById('fte_view');
-
-        assert.ok(!iframe.classList.contains('non-ready'));
-
-        // The second SIM has FTE passed
-        window.ConfigManager = new MockConfigManager({
-          fakeSettings: { fte: false },
-          applicationMode: applicationMode
+      function _finalizeFTE (evt) {
+        window.removeEventListener('ftestarted', _finalizeFTE);
+        window.ConfigManager.setOption({ fte: false }, function() {
+          window.postMessage({ type: 'fte_finished' }, Common.COST_CONTROL_APP);
         });
+      }
+      window.addEventListener('ftestarted', _finalizeFTE);
+
+      this.sinon.stub(Common, 'closeFTE', function() {
+        done();
+      });
 
-        window.addEventListener('fteClosed', function checkAssertions() {
-          window.removeEventListener('fteClosed', checkAssertions);
-            iframe = document.getElementById('fte_view');
-            assert.ok(iframe.classList.contains('non-ready'));
+      CostControlApp.init();
+  });
 
-            done();
+  test('UpdateUI always execute the callback function without a mode change',
+    function(done) {
+      var applicationMode = null;
```

Comment:
Initialization for this and the previous test is almost the same. Could you provide common functions to perform parametization? I would create a suite for FTE (you can choose to write a nested suite if you want).